File: patch_app.py
import sys
with open('src/core/app.js', 'r', encoding='utf-8') as f:
    content = f.read()

target = """              <!-- Model vs Kalshi probability — the primary insight -->
              <div style="margin-top:6px;padding:6px 8px;border-radius:4px;background:rgba(0,0,0,0.12);font-size:12px">
                <div style="display:flex;align-items:center;gap:8px;margin-bottom:${liveKPct != null ? '5px' : '0'}">
                  <span style="color:var(--color-text-muted);font-size:10px;font-weight:600;letter-spacing:.4px">MODEL</span>
                  <strong style="font-size:14px;color:${modelColor}">${modelLeanStr}</strong>
                  <span style="font-size:10px;font-weight:800;color:${modelDirectionColor};padding:1px 6px;border-radius:999px;background:rgba(125,183,255,0.10)">${modelLean === 'up' ? 'UP' : modelLean === 'down' ? 'DOWN' : 'WAIT'}</span>
                  <span style="color:var(--color-text-faint);font-size:16px;font-weight:300">↔</span>
                  <span style="color:var(--color-text-muted);font-size:10px;font-weight:600;letter-spacing:.4px">KALSHI</span>
                  <strong style="font-size:14px;color:${kalshiColor}">${kalshiLeanStr}</strong>
                  <span style="margin-left:auto;display:flex;align-items:center;gap:6px">
                    ${alignBadge}
                    ${edgePp != null ? `<span style="color:${edgePp >= 20 ? 'var(--color-green)' : edgePp >= 10 ? '#ffd700' : 'var(--color-text-faint)'};font-size:10px;font-weight:${edgePp >= 15 ? '800' : '600'}">${edgePp}pp${edgePp >= 20 ? ' ⚡' : edgePp >= 10 ? ' ▲' : ''}</span>` : ''}
                  </span>
                </div>"""

replacement = """              <!-- Model vs Kalshi probability — the primary insight -->
              <div style="margin-top:4px;padding:4px 6px;border-radius:4px;background:rgba(0,0,0,0.12);font-size:11px;overflow:hidden">
                <div style="display:flex;align-items:center;gap:4px;margin-bottom:${liveKPct != null ? '4px' : '0'};white-space:nowrap">
                  <span style="color:var(--color-text-muted);font-size:9px;font-weight:600;letter-spacing:.4px">MODEL</span>
                  <strong style="font-size:12px;color:${modelColor}">${modelLeanStr}</strong>
                  <span style="font-size:9px;font-weight:800;color:${modelDirectionColor};padding:1px 4px;border-radius:999px;background:rgba(125,183,255,0.10)">${modelLean === 'up' ? 'UP' : modelLean === 'down' ? 'DN' : 'WAIT'}</span>
                  <span style="color:var(--color-text-faint);font-size:12px;font-weight:300;margin:0 2px">↔</span>
                  <span style="color:var(--color-text-muted);font-size:9px;font-weight:600;letter-spacing:.4px">KALSHI</span>
                  <strong style="font-size:12px;color:${kalshiColor}">${kalshiLeanStr}</strong>
                  <span style="margin-left:auto;display:flex;align-items:center;gap:4px;overflow:hidden">
                    ${alignBadge}
                    ${edgePp != null ? `<span style="color:${edgePp >= 20 ? 'var(--color-green)' : edgePp >= 10 ? '#ffd700' : 'var(--color-text-faint)'};font-size:9px;font-weight:${edgePp >= 15 ? '800' : '600'}">${edgePp}pp</span>` : ''}
                  </span>
                </div>"""

target2 = """                <div style="display:grid;grid-template-columns:44px 1fr;row-gap:3px;align-items:center;font-size:9px">"""
rep2 = """                <div style="display:grid;grid-template-columns:36px 1fr;row-gap:3px;align-items:center;font-size:9px">"""

c1 = content.replace(target, replacement)
c2 = c1.replace(target.replace('\n', '\r\n'), replacement.replace('\n', '\r\n'))
c3 = c2.replace(target2, rep2)
c4 = c3.replace(target2.replace('\n', '\r\n'), rep2.replace('\n', '\r\n'))

with open('src/core/app.js', 'w', encoding='utf-8') as f:
    f.write(c4)

if c4 != content:
    print("Patched successfully")
else:
    print("Patch failed, target string not found")

File: patch_app.js
const fs = require('fs');
let content = fs.readFileSync('src/core/app.js', 'utf8');

// 1. Comment out the entire HOLD GUARD safely
const holdGuardStart = content.indexOf('// Safety gate: avoid late/borderline calls');
if (holdGuardStart !== -1) {
  const holdGuardEnd = content.indexOf('// ── Verdict', holdGuardStart);
  if (holdGuardEnd !== -1) {
    const originalGuard = content.substring(holdGuardStart, holdGuardEnd);
    const commentedGuard = '/* HOLD GUARD DISABLED BY USER REQUEST\n' + originalGuard + '*/\n    const confNorm = _normalizeConfidence(p.confidence) ?? 0;\n    ';
    content = content.replace(originalGuard, commentedGuard);
  }
}

// 2. Inject AI Verdict Override and Payout Guard
const displayVerdictTarget = "const displayVerdictDir = (\n      verdictDir === 'wait'";
const displayVerdictDirIndex = content.indexOf(displayVerdictTarget);
if (displayVerdictDirIndex !== -1) {
  const overrideLogic = `
    // --- AI VERDICT OVERRIDE ---
    if (p.llm?.ai_verdict?.direction) {
      const aiDir = p.llm.ai_verdict.direction.toLowerCase();
      if (aiDir === 'up' || aiDir === 'down' || aiDir === 'wait') {
        verdictDir = aiDir;
        if (aiDir === 'wait') {
          waitRationale = p.llm.ai_verdict.logic_recheck_summary || 'AI synthesized data and concluded WAIT is optimal.';
        }
      }
    }

    // --- PAYOUT/ODDS GUARD (USER REQUEST) ---
    if (_k15mV && verdictDir !== 'wait') {
      const yesPrice = _k15mV.yesAsk || _k15mV.yesBid || 50;
      if (verdictDir === 'up' && yesPrice >= 60) {
        verdictDir = 'wait';
        waitRationale = 'Payout too terrible (YES is at ' + Math.round(yesPrice) + '%). Edge is squeezed.';
      } else if (verdictDir === 'down' && yesPrice <= 40) {
        verdictDir = 'wait';
        waitRationale = 'Payout too terrible (NO is at ' + Math.round(100 - yesPrice) + '%). Edge is squeezed.';
      }
    }

    `;
  content = content.slice(0, displayVerdictDirIndex) + overrideLogic + content.slice(displayVerdictDirIndex);
}

// 3. Inject AI DIRECTIVE UI block
const rationaleTarget = '          ${waitRationale ? `<div class="pred-verdict-rationale">${waitRationale}</div>` : \'\'}';
if (content.includes(rationaleTarget)) {
  const replacement = `          \${(p.llm?.ai_wording?.wait_rationale) ? \`<div class="pred-verdict-rationale"><span style="font-weight:700">✨ AI:</span> \${escapeHtml(p.llm.ai_wording.wait_rationale)}</div>\` : waitRationale ? \`<div class="pred-verdict-rationale">\${waitRationale}</div>\` : ''}
          
          \${(verdictDir !== 'wait') ? \`
            <div class="ai-insights-block" style="margin-top:14px;margin-bottom:8px;padding:16px;border-radius:12px;background:\${verdictDir === 'up' ? 'rgba(38,212,126,0.1)' : 'rgba(255,68,102,0.1)'};border:2px solid \${verdictDir === 'up' ? 'rgba(38,212,126,0.4)' : 'rgba(255,68,102,0.4)'};box-shadow: 0 4px 15px rgba(0,0,0,0.15);">
              <div style="font-size:18px;font-weight:900;color:\${verdictDir === 'up' ? 'var(--color-green)' : 'var(--color-red)'};margin-bottom:10px;text-transform:uppercase;letter-spacing:1px;display:flex;align-items:center;justify-content:space-between">
                <span>⚡ AI DIRECTIVE: BUY \${verdictDir === 'up' ? 'YES' : 'NO'}</span>
              </div>
              \${(p.llm?.ai_wording?.primary_rationale) ? \`<div style="font-size:14px;color:rgba(255,255,255,0.85);font-style:italic;padding-left:10px;border-left:3px solid \${verdictDir === 'up' ? 'var(--color-green)' : 'var(--color-red)'}">"\${escapeHtml(p.llm.ai_wording.primary_rationale)}"</div>\` : ''}
              \${(p.llm?.ai_wording?.high_confidence_rationale) ? \`<div style="font-size:14px;color:rgba(255,255,255,0.85);font-style:italic;padding-left:10px;border-left:3px solid \${verdictDir === 'up' ? 'var(--color-green)' : 'var(--color-red)'};margin-top:6px">"\${escapeHtml(p.llm.ai_wording.high_confidence_rationale)}"</div>\` : ''}
            </div>
          \` : ''}`;
  content = content.replace(rationaleTarget, replacement);
}

fs.writeFileSync('src/core/app.js', content);
console.log('App patched successfully');
